include/wx/gtk/caret.h

```diff
@@ -0,0 +1,108 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/gtk/caret.h
+// Purpose:     wxCaret class - the GTK3 implementation of wxCaret
+// Author:      Kettab Ali
+// Created:     2022-01-29
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_CARET_H_
+#define _WX_CARET_H_
+
+#include "wx/timer.h"
+
+class WXDLLIMPEXP_CORE wxCaret : public wxCaretBase
+{
+public:
+    wxCaret() : m_timer(this) { Init(); }
+        // create the caret of given (in pixels) width and height and associate
+        // with the given window
+    wxCaret(wxWindow *window, int width, int height) : m_timer(this)
+    {
+        Init();
+
+        (void)Create(window, width, height);
+    }
+        // same as above
+    wxCaret(wxWindowBase *window, const wxSize& size) : m_timer(this)
+    {
+        Init();
+
+        (void)Create(window, size);
+    }
+
+    // process wxWindow notifications
+    virtual void OnSetFocus() wxOVERRIDE;
+    virtual void OnKillFocus() wxOVERRIDE;
+
+protected:
+    void Init()
+    {
+        wxCaretBase::Init();
+
+        if ( GetBlinkTime() == 1200 )
+        {
+            // for MSW compatibility, map gtk-cursor-blink-time
+            // default value '1200' to '500'.
+            SetBlinkTime(500);
```

Comment:
I don't think we should do this, there is no need for a caret under Linux to be compatible with the caret under MSW.